### DISCUSSION TITLE
Windows Server, version 1803 no longer supported by .NET Core

### DISCUSSION THREAD

**Discussion by mthalman** [dotnet/dotnet-docker#3661] [Announcements]
# Windows Server, version 1803 no longer supported by .NET Core

Windows Server, version 1803 is no longer supported. As a result, this version of Windows is no longer supported by the .NET Team and we will no longer produce .NET Core container images for this Windows version.

Support statements:

* [Currently supported .NET Core Docker images](https://hub.docker.com/_/microsoft-dotnet-core)
* [.NET Core Support Policy](https://dotnet.microsoft.com/platform/support/policy/dotnet-core)
* [Products Reaching End of Support for 2019](https://support.microsoft.com/en-us/help/4316957/products-reaching-end-of-support-for-2019)

## Details

Docker tags that contain `nanoserver-1803` are no longer being produced. We will not delete existing images or those tags. However, they contain unsupported builds of Nano Server and .NET Core.

## Pulling manifest tags

These changes also affect manifest tags (AKA “multi-arch tags“), such as `3.0`, `3.0.0`. These are tags that are supported on multiple operating systems, and do not contain the operating system in them. As a result of no longer producing new images for Windows Server, version 1803, manifest tags no longer include entries for Windows 10, version 1803, and Windows Server, version 1803. That means that .NET Core manifest tags are no longer usable on those Windows versions.

If you pull a manifest tag on one of the Windows versions described, you may get one of the following error messages:

* `a Windows version 10.0.18362-based image is incompatible with a 10.0.17134 host`
* `no matching manifest for windows/amd64 in the manifest list entries`
* `no matching manifest for unknown in the manifest list entries`

If you need to still use Nano Server, version 1803 images (even though it is no longer supported), you will need to use tags that include the Nano Server version (these are non-manifest tags), for example `mcr.microsoft.com/dotnet/core/runtime:3.0-nanoserver-1803`.

**Comment by lioobayoyo** [comment]
I'm getting this error while trying to build the sample app aspnet mentioned in the docs with `docker build  :
```
PS C:\source\repos\dotnet-docker\samples\aspnetapp> docker build .
Sending build context to Docker daemon  9.082MB
Step 1/12 : FROM mcr.microsoft.com/dotnet/core/sdk:3.0 AS build
3.0: Pulling from dotnet/core/sdk
no matching manifest for windows/amd64 10.0.17134 in the manifest list entries
```
It seems my working Windows (Professional, not Server) is this 1803. (I'm getting this information when using `docker info`).

What am I supposed to write in the Dockerfile (which is source control, that may be used by other developers with a different windows versions, or deployed on another environment altogether).

As I understand, it's not a good to replace the from by `FROM mcr.microsoft.com/dotnet/core/sdk:3.0-nanoserver-1803` in the Dockerfile, right ?

**Comment by lioobayoyo** [comment]
note: I'm having same error with the example in the windows readme file : 
```
docker run --name aspnetcore_sample --rm -it -p 8000:80 mcr.microsoft.com/dotnet/core/samples:aspnetapp
```

So, my general question is, **how can I develop and test locally with docker with this Windows version** ? (which is set up by my company, I don't have any possibility whatsoever to change it)

**Comment by mthalman** [comment]
> As I understand, it's not a good to replace the from by FROM mcr.microsoft.com/dotnet/core/sdk:3.0-nanoserver-1803 in the Dockerfile, right ?

That's not necessarily true.  The tag is still available.  The manifest tag for `3.0` no longer has an entry for `3.0-nanoserver-1803` due to end of support for 1803 which is why you're seeing the `no matching manifest for windows/amd64 10.0.17134 in the manifest list entries` error.

I'm curious, what is the context of your company's usage of Docker image?  Are you merely consuming this base image or are you publishing your own?

If you're publishing them, presumably the host Docker machine that produces your official builds is on a different OS version than 1803, otherwise it would be getting the same manifest error.  And it wouldn't make sense to be publishing an image based on an out-of-support OS version.

If everyone that's consuming this Dockerfile is using 1803, then yes it would make sense to use the `3.0-nanoserver-1803` tag to become unblocked.

**Comment by lioobayoyo** [comment]
Late answer, but just to not let this unresponded :
in my case, I'm a developer, aiming to push images into our pivate repository for deploying the software we produce in our various environments (cloud / VM based, most often). The specific Windows version actually makes no sense at all, indeed.
For the record, switching to Linux containers makes everything works smooth and easy.
